src/extension.ts

```diff
@@ -75,30 +72,10 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
         },
         connect: async (connectionInfo: IConnectionInfo, saveConnection?: boolean) => {
             const uri = utils.generateQueryUri().toString();
-            const connectionPromise = new Deferred<boolean>();
             // First wait for initial connection request to succeed
-            const requestSucceeded = await controller.connect(
-                uri,
-                connectionInfo,
-                connectionPromise,
-                saveConnection,
-            );
+            const requestSucceeded = await controller.connect(uri, connectionInfo, saveConnection);
             if (!requestSucceeded) {
-                if (controller.connectionManager.failedUriToFirewallIpMap.has(uri)) {
```

Comment:
We don't need special handling for errors as the connect call already does that now.